src/registrar/views/portfolio_members_json.py

```diff
@@ -86,24 +104,42 @@ def initial_permissions_search(portfolio):
             "email_display",
             "last_active",
             "roles",
-            "additional_permissions_display",
+            "additional_permissions",
             "member_display",
+            "domain_info",
             "source",
         )
     )
     return permissions
 
 
+# Custom Func to use array_remove to remove null values
+class ArrayRemove(Func):
+    function = "array_remove"
+    template = "%(function)s(%(expressions)s, NULL)"
+
+
 def initial_invitations_search(portfolio):
-    """Perform initial invitations search before applying any filters."""
+    """Perform initial invitations search and get related DomainInvitation data based on the email."""
+    # Get DomainInvitation query for matching email
+    domain_invitations = DomainInvitation.objects.filter(email=OuterRef("email")).annotate(
+        domain_info=Concat(F("domain__id"), Value(":"), F("domain__name"), output_field=CharField())
+    )
+    # PortfolioInvitation query
     invitations = PortfolioInvitation.objects.filter(portfolio=portfolio)
     invitations = invitations.annotate(
         first_name=Value(None, output_field=CharField()),
         last_name=Value(None, output_field=CharField()),
         email_display=F("email"),
         last_active=Value("Invited", output_field=TextField()),
-        additional_permissions_display=F("additional_permissions"),
         member_display=F("email"),
+        # Use ArrayRemove to return an empty list when no domain invitations are found
```

Comment:
This was actually pretty strange and unexpected results when using subqueries. A list with a single 'null' entry was being returned. Had to explicitly remove the null entry to return an empty list. Otherwise, the null value was being processed as an item in the list, and each time traversing the list, had to check for null values and had to handle exceptional case. Removing the null in the query returned expected results (ie. empty list when no found values) to make later code more readable and predictable.